autoPyTorch/api/base_task.py

```diff
@@ -368,6 +369,7 @@ def get_dataset(
         resampling_strategy_args: Optional[Dict[str, Any]] = None,
         dataset_name: Optional[str] = None,
         dataset_compression: Optional[DatasetCompressionSpec] = None,
+        adaptive_memory_allocation: bool = True
```

Comment:
I think its better if we allow the users to pass `data_compression = {'memory_allocation': 'adaptive' or '0.1'(example), 'methods': ['precision', 'subsample']}`. We can also do this when we finally merge it to the dev branch. Let me know if you would like to do it for this PR.

Reply:
I will adapt the code to your suggestion quickly

Reply:
yeah and you can make `adaptive` as the default value